07.ls_object/ls.rb

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env ruby
+
+# frozen_string_literal: true
+
+require './ls_file'
+require 'optparse'
+
+class LS
+  MAX_ROW = 3
+
+  def main
+    opt = OptionParser.new
+
+    params = { dot_match: false, long_format: false, reverse: false }
+    opt.on('-a') { |v| params[:dot_match] = v }
+    opt.on('-l') { |v| params[:long_format] = v }
+    opt.on('-r') { |v| params[:reverse] = v }
+    opt.parse!(ARGV)
+    path = '.'
+
+    collected_filenames = collect_files(path, params[:dot_match], params[:reverse])
+    puts params[:long_format] ? list_long(collected_filenames) : list_short(collected_filenames)
+  end
+
+  private
+
+  def collect_files(path, dot_match, reverse)
+    all_filenames = Dir.foreach(path)
+    filenames = dot_match ? all_filenames : all_filenames.reject { |file| file.start_with?('.') }
+    sorted_filenames = sort_filenames(filenames)
+    reverse ? sorted_filenames.reverse : sorted_filenames
+  end
+
+  def sort_filenames(filenames)
+    filenames.sort do |a, b|
+      a = a.slice(1..-1) if a.start_with?('.') && !['.', '..'].include?(a)
+      b = b.slice(1..-1) if b.start_with?('.') && !['.', '..'].include?(b)
+      a.downcase <=> b.downcase
+    end
+  end
+
+  def list_long(filenames)
+    ls_file_objs = filenames.map { |filename| LSFile.new(filename) }
```

Comment:
```suggestion
    ls_files = filenames.map { |filename| LSFile.new(filename) }
```

シンプルに複数形にして配列であることを表現すると良いと思います

Reply:
修正しました